src/transports/metaplaylist/pipelines.ts

```diff
@@ -174,55 +172,47 @@ export default function(options : ITransportOptions): ITransportPipelines {
-        if (parsedResult.type === "done") {
-          const manifest = new Manifest(parsedResult.value, options);
-          return observableOf({ manifest });
-        }
-
-        const loaders$ : Array<Observable<Manifest>> =
-          parsedResult.value.ressources.map((ressource) => {
-            const transport = getTransportPipelines(transports,
-                                                    ressource.transportType,
-                                                    otherTransportOptions);
-            const request$ = scheduleRequest(() =>
-                transport.manifest.loader({ url : ressource.url }).pipe(
-                  filter((e): e is ILoaderDataLoaded< Document | string > =>
-                    e.type === "data-loaded"
-                  ),
-                  map((e) : ILoaderDataLoadedValue< Document | string > => e.value)
-                ));
-
-            return request$.pipe(mergeMap((responseValue) => {
-              return transport.manifest.parser({ response: responseValue,
-                                                 url: ressource.url,
-                                                 scheduleRequest,
-                                                 externalClockOffset })
-                .pipe(map((parserData) : Manifest => parserData.manifest));
-            }));
-          });
+      const parserOptions = { url,
+                              serverSyncInfos: options.serverSyncInfos };
 
-        return combineLatest(loaders$).pipe(mergeMap((loadedRessources) =>
-          handleParsedResult(parsedResult.value.continue(loadedRessources))
-        ));
-      }
+      const parsedManifest = parseMetaPlaylist(responseData, parserOptions);
+      const manifest = new Manifest(parsedManifest, options);
+      return observableOf({ manifest });
+    },
+  };
+
+  const periodPipeline = {
+    loader({ period } : IPeriodLoaderArguments) : IPeriodLoaderObservable {
+      const url = getURLFromPeriod(period);
+      const pipelines = getTransportPipelinesFromPeriod(period);
+      return pipelines.manifest.loader({ url });
+    },
+
+    parser({ externalClockOffset, // XXX TODO
+             period,
+             response,
+             scheduleRequest } : IPeriodParserArguments) : IPeriodParserObservable {
+      const url = getURLFromPeriod(period);
+      const pipelines = getTransportPipelinesFromPeriod(period);
+      return pipelines.manifest.parser({ response,
+                                         url,
+                                         scheduleRequest,
+                                         externalClockOffset })
+        .pipe(map((parserData) => {
+          const transformed =
+            transformManifestToMetaplaylistPeriod(period, parserData.manifest);
+          const periods = transformed.map(parsedPeriod => {
+            return parsedPeriod.isLoaded ? new LoadedPeriod(parsedPeriod) :
+                                           new PartialPeriod(parsedPeriod);
```

Comment:
TODO what about the representationFilter.

Reply:
We will have issues if we're going this way with Representation in HLS.
As the representation filter would not be able to be run there.

Plus we will have to handle a refused `LoadedRepresentation` than an AdaptationBuffer has chosen (through the ABRMamager). I don't know yet how we will do that. But there is another refacto here that should be thought of.